src/Hl7.Fhir.Core/Serialization/ElementNavFhirReader.cs

```diff
@@ -123,8 +120,31 @@ private bool verifyXmlSpecificDetails(IElementNavigator child)
                 throw Error.Format($"Xml node of type '{xmlDetails.NodeType}' is unexpected at this point", this);
         }
 
+        public bool MoveToNext(string nameFilter = null)
+        {
+            return _current.MoveToNext(nameFilter);
+        }
+
+        public bool MoveToFirstChild(string nameFilter = null)
+        {
+            return _current.MoveToFirstChild(nameFilter);
+        }
+
+        public IElementNavigator Clone()
+        {
+            return _current.Clone();
```

Comment:
The Clone should return a clone ElementNavFhirReader instead of the inner reader.

Reply:
Yes, it should indeed.

Reply:
Turned ElementNavReader into a struct, create a clone of itself by creating a new instance + clone the inner IElementNavigator (_current)